module/zfs/arc.c

```diff
@@ -1764,25 +1757,15 @@ void
 arc_buf_freeze(arc_buf_t *buf)
 {
 	arc_buf_hdr_t *hdr = buf->b_hdr;
-	kmutex_t *hash_lock;
 
 	if (!(zfs_flags & ZFS_DEBUG_MODIFY))
 		return;
 
-	if (ARC_BUF_COMPRESSED(buf)) {
-		ASSERT(hdr->b_l1hdr.b_freeze_cksum == NULL ||
-		    arc_hdr_has_uncompressed_buf(hdr));
+	if (ARC_BUF_COMPRESSED(buf))
 		return;
-	}
-
-	hash_lock = HDR_LOCK(hdr);
-	mutex_enter(hash_lock);
 
 	ASSERT(HDR_HAS_L1HDR(hdr));
```

Comment:
For non-debug builds.  Let's drop the local variable which is only used in the ASSERT now.

```
../../module/zfs/arc.c: In function 'arc_buf_freeze':
../../module/zfs/arc.c:1759:17: warning: unused variable 'hdr' [-Wunused-variable]
```